sdk/github.md

```diff
@@ -83,14 +83,43 @@ This is the name of the file within the `.github/workflows` folder.
   - `assemble`:
     - build the project (archives / artifacts)
     - for Gradle Java projects this is `./gradlew assemble`
-  - `check` - for verification tasks, including:
+    - pushes artifacts, where logical, to `sdk.ably.com` (staging)
+  - `check` - for basic, quick verification tasks, including:
     - run linters
     - perform static analysis if available
     - run unit tests
     - generate code coverage reports from unit test runs if possible (probably uploaded as [artifacts](https://docs.github.com/en/actions/advanced-guides/storing-workflow-data-as-artifacts))
     - for Gradle Java projects this is `./gradlew check`
+    - should not run time consuming integration tests
+    - examples:
+      [ably-asset-tracking-android](https://github.com/ably/ably-asset-tracking-android/blob/main/.github/workflows/check.yml),
+      [ably-java](https://github.com/ably/ably-java/blob/main/.github/workflows/check.yml),
+      [ably-js](https://github.com/ably/ably-js/blob/main/.github/workflows/check.yml),
+      [engineering](https://github.com/ably/engineering/blob/main/.github/workflows/check.yaml)
+  - `docs`:
+    - build API reference documentation from interface commentary
+    - should push artifacts to `sdk.ably.com` (staging)
+    - examples:
+      [ably-asset-tracking-android](https://github.com/ably/ably-asset-tracking-android/blob/main/.github/workflows/docs.yml),
+      [ably-asset-tracking-swift](https://github.com/ably/ably-asset-tracking-swift/blob/main/.github/workflows/docs.yml),
+      [ably-flutter](https://github.com/ably/ably-flutter/blob/main/.github/workflows/docs.yml)
   - `emulate`:
+    - runs tests, typically time consuming integration tests
+    - spawns an emulator to do the testing - typically Android or iOS
     - for Gradle Java projects this is `./gradlew connectedCheck`
+    - example:
+      [ably-asset-tracking-android](https://github.com/ably/ably-asset-tracking-android/blob/main/.github/workflows/emulate.yml)
+  - `integration-test`:
```

Comment:
I have an issue with this - in ably-js we have integration tests running on multiple platforms (browser and node at the moment, possibly react-native in the future). It's pretty common that I'll want to re-run the integration tests on a particular platform but if I put them in to one workflow file then I'll only be able to re-run every single integration test. Also these tests take a long time and I'm a bit unsure of whether it will be easily possible to run both the browser and node integration tests in parallel with a single workflow file.

My proposed solution to this is to allow for all of these workflow names an arbitrary suffix delimited by a `:` character, for example `integration-test:node` and `integration-test:browser`. WDYT?

Reply:
I like that. I'll write something up around that.

Reply:
Actually, @owenpearson, now I get around to writing this up... I think a colon `':'` may not be the best delimiter to choose, especially taking Windows users into account. Perhaps we can just stick to dash `'-'`. 🤔 ... e.g. `integration-test-node.yaml`

Reply:
I just checked and `:` isn't allowed on macOS either 😅 

It would be nice to have a different delimiter but all of the good ones are disallowed by Windows so yes, I think sticking with the dash is our best option.

Reply:
https://github.com/ably/engineering/pull/13/commits/313966b7168fd2d87b39d0f70f05bb883db7102e